gcc/config/riscv/riscv.c

```diff
@@ -3085,7 +3085,16 @@ riscv_compute_frame_info (void)
 
       /* Only use save/restore routines if they don't alter the stack size.  */
       if (RISCV_STACK_ALIGN (num_save_restore * UNITS_PER_WORD) == x_save_size)
-	frame->save_libcall_adjustment = x_save_size;
+	{
+	  frame->save_libcall_adjustment = x_save_size;
+	  if (TARGET_RVE && riscv_use_save_libcall (frame))
+	    {
+	      /* Libcall save/restore 3 registers at once, so we need
+		 allocate 12 bytes for callee-saved register.  */
+	      x_save_size = 12;
+	      frame->save_libcall_adjustment = x_save_size;
+	    }
```

Comment:
I think this is clearer:

 	  if (TARGET_RVE && riscv_use_save_libcall (frame))
 	    {
 	      /* Libcall saves/restores 3 registers at once, so we need
 		 allocate 12 bytes for callee-saved register.  */
 	      x_save_size = 3 * UNITS_PER_WORD;
 	    }
	  frame->save_libcall_adjustment = x_save_size;